Rename of module does not handle import aliases correctly

I am using rope to rename modules in my project and came across a problem with how it handles aliases. Here is a minimal example that shows this behavior. In this case, `mod2.py` imports `mod1.py` and so should import `mod1.py` with a new name when `mod1.py`'s name is changed. However, in this case no changes are actually made in `mod2.py`. I am using version 0.10.7.

Here is my example:
```python
from rope.refactor.rename import Rename
from rope.base.project import Project

project = Project('.')

mod1 = project.root.create_file('mod1.py')
mod1.write(
    'class Test(object):\n'
    '    def __init__(self):'
    '        print "hello world"')

mod2 = project.root.create_file('mod2.py')
mod2string = 'import mod1 as m\n' + 'm.Test()'
mod2.write(mod2string)

new_name = 'AAAAAAAA'
changes = Rename(project, mod1).get_changes(new_name)
project.do(changes)

expected = 'import ' + new_name + ' as m\n' + 'm.Test()'
with open('mod2.py', 'r') as f:
    actual = f.read()
print '~' * 50
print actual
print '~' * 50
print expected
assert  actual == expected

```
I get the following output
```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import mod1 as m
m.Test()
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import AAAAAAAA as m
m.Test()

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-52-b92f47d557f0> in <module>()
     25     print '~' * 50
     26     print expected
---> 27     assert  actual == expected
     28 

AssertionError: 
```

--- a/rope/base/evaluate.py
+++ b/rope/base/evaluate.py
@@ -101,8 +101,9 @@ class ScopeNameFinder(object):
                                                      offset, lineno):
             name = self.worder.get_primary_at(offset).strip()
             return (None, holding_scope.parent[name])
-        # from statement module
-        if self.worder.is_from_statement_module(offset):
+        # module in a from statement or an imported name that is aliased
+        if (self.worder.is_from_statement_module(offset) or
+            self.worder.is_import_statement_aliased_module(offset)):
             module = self.worder.get_primary_at(offset)
             module_pyname = self._find_module(module)
             return (None, module_pyname)

--- a/rope/base/worder.py
+++ b/rope/base/worder.py
@@ -65,6 +65,9 @@ class Worder(object):
     def is_from_aliased(self, offset):
         return self.code_finder.is_from_aliased(offset)
 
+    def is_import_statement_aliased_module(self, offset):
+        return self.code_finder.is_import_statement_aliased_module(offset)
+
     def find_parens_start_from_inside(self, offset):
         return self.code_finder.find_parens_start_from_inside(offset)
 
@@ -317,7 +320,9 @@ class _RealFinder(object):
             last_import = self.code.rindex('import ', 0, offset)
         except ValueError:
             return False
-        return self._find_import_end(last_import + 7) >= offset
+        line_start = self._get_line_start(last_import)
+        return (self._find_import_end(last_import + 7) >= offset and
+                self._find_word_start(line_start) == last_import)
 
     def is_from_statement(self, offset):
         try:
@@ -336,6 +341,27 @@ class _RealFinder(object):
         line_start = self._get_line_start(stmt_start)
         prev_word = self.code[line_start:stmt_start].strip()
         return prev_word == 'from'
+
+    def is_import_statement_aliased_module(self, offset):
+        if not self.is_import_statement(offset):
+            return False
+        try:
+            line_start = self._get_line_start(offset)
+            import_idx = self.code.rindex('import', line_start, offset)
+            imported_names = import_idx + 7
+        except ValueError:
+            return False
+        # Check if the offset is within the imported names
+        if (imported_names - 1 > offset or
+            self._find_import_end(imported_names) < offset):
+            return False
+        try:
+            end = self._find_word_end(offset)
+            as_end = min(self._find_word_end(end + 1), len(self.code))
+            as_start = self._find_word_start(as_end)
+            return self.code[as_start:as_end + 1] == 'as'
+        except ValueError:
+            return False
 
     def is_a_name_after_from_import(self, offset):
         try:
@@ -368,8 +394,7 @@ class _RealFinder(object):
             end = self._find_word_end(offset)
             as_end = min(self._find_word_end(end + 1), len(self.code))
             as_start = self._find_word_start(as_end)
-            if self.code[as_start:as_end + 1] == 'as':
-                return True
+            return self.code[as_start:as_end + 1] == 'as'
         except ValueError:
             return False
 

--- a/ropetest/refactor/renametest.py
+++ b/ropetest/refactor/renametest.py
@@ -223,6 +223,16 @@ class RenameRefactoringTest(unittest.TestCase):
                         self.project.find_module('newmod') is not None)
         self.assertEquals('from newmod import a_func\n', mod2.read())
 
+    def test_renaming_modules_aliased(self):
+        mod1 = testutils.create_module(self.project, 'mod1')
+        mod1.write('def a_func():\n    pass\n')
+        mod2 = testutils.create_module(self.project, 'mod2')
+        mod2.write('import mod1 as m\nm.a_func()\n')
+        self._rename(mod1, None, 'newmod')
+        self.assertTrue(not mod1.exists() and
+                        self.project.find_module('newmod') is not None)
+        self.assertEquals('import newmod as m\nm.a_func()\n', mod2.read())
+
     def test_renaming_packages(self):
         pkg = testutils.create_package(self.project, 'pkg')
         mod1 = testutils.create_module(self.project, 'mod1', pkg)

